.github/workflows/windows_cmake.yml

```diff
@@ -0,0 +1,160 @@
+      shell: bash
+      run: |
+        echo "LIBT_DEPR_FUN=OFF" >> $GITHUB_ENV
+
+    - name: setup environment - don't build tests
+      if: matrix.build_tests == 'notests'
+      shell: bash
+      run: |
+        echo "LIBT_TESTS=OFF" >> $GITHUB_ENV
+
+    - name: setup environment - don't build with python bindings
+      if: matrix.python_bindings == 'nopy'
+      shell: bash
+      run: |
+        echo "LIBT_PY_BINDINGS=OFF" >> $GITHUB_ENV
+
+    # NOTE: MSVC tools must be in the path for the Ninja generator. Caveat: must use cmd
+    - name: setup MSVC dev cmd
+      uses: ilammy/msvc-dev-cmd@v1.4.1
+
+    # ninja is preferable, not a hard requirement
+    - name: install additional required packages with chocolatey
+      run: |
+        choco install ninja
+
+    - name: setup vcpkg (cached, if possible)
+      uses: lukka/run-vcpkg@v4.1
+      with:
+        vcpkgDirectory: ${{ env.VCPKG_DEST_WIN }}
+        vcpkgGitCommitId: ${{ env.VCPKG_COMMIT }}
+        setupOnly: true
+        appendedCacheKey: ${{ matrix.build_variant }}
+
+    # clear buildtrees to reduce disk space requirements
+    - name: install dependencies via vcpkg
+      run: |
+        ${{ env.RUNVCPKG_VCPKG_ROOT }}/vcpkg.exe install `
+          boost-asio:${{ env.VCPKG_TRIPLET }} `
+          boost-chrono:${{ env.VCPKG_TRIPLET }} `
+          boost-config:${{ env.VCPKG_TRIPLET }} `
+          boost-crc:${{ env.VCPKG_TRIPLET }} `
+          boost-date-time:${{ env.VCPKG_TRIPLET }} `
+          boost-iterator:${{ env.VCPKG_TRIPLET }} `
+          boost-multiprecision:${{ env.VCPKG_TRIPLET }} `
+          boost-pool:${{ env.VCPKG_TRIPLET }} `
+          boost-python:${{ env.VCPKG_TRIPLET }} `
+          boost-random:${{ env.VCPKG_TRIPLET }} `
+          boost-scope-exit:${{ env.VCPKG_TRIPLET }} `
+          boost-system:${{ env.VCPKG_TRIPLET }} `
+          boost-variant:${{ env.VCPKG_TRIPLET }} `
+          openssl:${{ env.VCPKG_TRIPLET }} `
+          --clean-after-build
+
+    - name: build libtorrent
+      shell: cmd
+      run: |
+        cmake -B cmake-build-dir -G "Ninja" ^
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=${{ matrix.build_config }} ^
+          -DCMAKE_TOOLCHAIN_FILE=${{ env.VCPKG_DEST_WIN }}\scripts\buildsystems\vcpkg.cmake ^
+          -DVCPKG_TARGET_TRIPLET=${{ env.VCPKG_TRIPLET }} ^
+          -Dbuild_examples=ON -Dbuild_tools=ON ^
+          -Dbuild_tests=${{ env.LIBT_TESTS }} -Ddeprecated-functions=${{ env.LIBT_DEPR_FUN }} ^
+          -DBUILD_SHARED_LIBS=${{ env.LIBT_BSL }} -Dstatic_runtime=${{ env.LIBT_STATIC_RT }} ^
+          -Dpython-bindings=${{ env.LIBT_PY_BINDINGS }} -Dskip-python-runtime-test=ON ^
+          -Dboost-python-module-name="python38" ^
+          --graphviz=cmake-build-dir\target_graph.dot
+        cmake --build cmake-build-dir
+
+    - name: run ctest
+      if: matrix.build_tests == 'tests'
+      run: |
+        cd cmake-build-dir && ctest --output-on-failure
```

Comment:
are there multiple CPUs on github's CI machines? if so, running the test might be faster by specifying `-j <n>` to `ctest`. But it seems some of the tests that require spinning up a python helper process are failing on windows.

On appveyor not all tests are run on windows, just the "deterministic_tests", which are ones that don't depend on global system resources like sockets and external python processes.

Reply:
@arvidn 

> are there multiple CPUs on github's CI machines? if so, running the test might be faster by specifying `-j <n>` to `ctest`. But it seems some of the tests that require spinning up a python helper process are failing on windows.

They have 2 CPUs. fixed.

> On appveyor not all tests are run on windows, just the "deterministic_tests", which are ones that don't depend on global system resources like sockets and external python processes.

How can I setup ctest to not run the problematic tests? Is the `-E` option a good solution, in your opinion?

Reply:
I think `-E test_upnp` is reasonable. It's not entirely clear to me that it's necessary though. It works on windows for me locally. I suspect that maybe appveyor VMs have some configuration that breaks multicast or multicast loopback.